rllib/utils/nested_dict.py

```diff
@@ -85,7 +85,7 @@ class NestedDict(Generic[T], MutableMapping[str, Union[T, "NestedDict"]]):
             >>>                             #            'b': {'c': 200, 'd': 300}}
             >>> # Getting elements, possibly nested:
             >>> print(foo_dict['b', 'c'])   # 200
-            >>> print(foo_dict['b']) # IndexError("Use get for partial indexing.")
+            >>> print(foo_dict['b']) # IndexError
```

Comment:
Dumb question: Is there a reason why we make this seemingly arbitrary distinction between accessing by `get` (no error) vs direct (error)? Why should both not return the sub-dict? How would the user know this difference?

Reply:
good point @sven1977, In hindsight I don't see any reason for not returning the sub-nested dict if `__getitem__` is used. It actually confused myself at some point during a later pr.

Reply:
fixed.